SWADroid/src/main/java/es/ugr/swad/swadroid/modules/games/MatchesActiveList.java

```diff
@@ -0,0 +1,296 @@
+
+import android.annotation.TargetApi;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.database.Cursor;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Handler;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+import android.util.Log;
+import android.view.View;
+import android.widget.AbsListView;
+import android.widget.AdapterView;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import java.lang.ref.WeakReference;
+
+/**
+ * Matches List module.
+ *
+ * @author Juan Miguel Boyero Corral <juanmi1982@gmail.com>
+ * @author Sergio Díaz Rueda <sergiodiazrueda8@gmail.com>
+ */
+public class MatchesActiveList extends MenuExpandableListActivity
+        implements SwipeRefreshLayout.OnRefreshListener {
+    /**
+     * MatchesActiveList tag name for Logcat
+     */
+    private static final String TAG = Constants.APP_TAG + " MatchesActiveList";
+
+    /**
+     * ListView of matches
+     */
+    private static ListView lvMatches;
+    /**
+     * Adapter for ListView of matches
+     */
+    private static GamesCursorAdapter adapter;
+    /**
+     * Handler of matches
+     */
+    private final MatchesActiveList.RefreshAdapterHandler mHandler =
+            new MatchesActiveList.RefreshAdapterHandler(this);
+
+    private final Runnable mRunnable = new Runnable() {
+        @Override
+        public void run() {
+      /*
+    Database cursor for Adapter of matches
+   */
+            Cursor dbCursor = dbHelper.getMatchesGameCursor(gameCode);
+            startManagingCursor(dbCursor);
+
+
+            /*
+             * If there aren't matches to show, hide the matches lvMatches
+             * and show the empty matches message
+             */
+            if ((dbCursor == null) || (dbCursor.getCount() == 0)) {
+                Log.d(TAG, "Matches list is empty");
+
+                emptyMatchesTextView.setText(R.string.matchesEmptyListMsg);
+                emptyMatchesTextView.setVisibility(View.VISIBLE);
+
+                lvMatches.setVisibility(View.GONE);
+            } else {
+                Log.d(TAG, "Matches list is not empty");
+
+                emptyMatchesTextView.setVisibility(View.GONE);
+                lvMatches.setVisibility(View.VISIBLE);
+            }
+
+            adapter = new GamesCursorAdapter(getBaseContext(), dbCursor, dbHelper);
+            lvMatches.setAdapter(adapter);
+
+            mProgressScreen.hide();
+        }
+    };
+    /**
+     * TextView for the empty matches message
+     */
+    private TextView emptyMatchesTextView;
+    /**
+     * Layout with "Pull to refresh" function
+     */
+    private SwipeRefreshLayout refreshLayout;
+    /**
+     * Progress screen
+     */
+    private ProgressScreen mProgressScreen;
+    /**
+     * ListView click listener
+     */
+    private ListView.OnItemClickListener clickListener = new ListView.OnItemClickListener() {

```

Comment:
Maybe you could move the `clickListener` initialization code to a private method to improve the code readability.